Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCalibration.py

```diff
@@ -35,152 +37,141 @@ def name(self):
-            if calib_file_error:
-                issues[self._CALIBRATION_FILE] = calib_file_error
         return issues
 
-    def _validate_calibration_filepath(self, filepath):
-        try:
-            self._scanned_pixel_positions = self._get_pixel_positions_from_file(filepath)
-            if len(self._scanned_pixel_positions) == 0:
-                return "Calibration file provided contains no data"
-        except FileNotFoundError:
-            return "Calibration file path cannot be found"
-        except Exception:
-            return "Could not read contents of calibration file, please see documentation to check the required format"
-        return ""
-
-    def _find_specular_pixel_index(self, ws, det_info):
-        # Determine position of the specular pixel from the workspace:
-        ws_two_theta = ws.run().getTimeAveragedValue("theta") * 2
-
-        min_theta_diff = None
-        specular_pixel_idx = None
-
-        for i in range(0, ws.getNumberHistograms()):
-            try:
-                detector = ws.getDetector(i)
-            except RuntimeError:
-                # Exclude point detectors that don't have IDs
-                continue
-
-            detector_index = det_info.indexOf(detector.getID())
-
-            if det_info.isMonitor(detector_index):
+    def _parse_calibration_file(self, filepath):
+        """Create a dictionary of detector IDs and theta offsets from the calibration file"""
+        scanned_theta_offsets = {}
+        with open(filepath, "r") as file:
+            labels_checked = False
+
+            file_reader = csv.reader(file)
+            for row in file_reader:
+                if len(row) == 0:
+                    # Ignore any blank lines
+                    continue
+
+                entries = row[0].split()
+
+                if entries[0][0] == "#":
+                    # Ignore any lines that begin with a #
+                    # This allows the user to add any metadata they would like
+                    continue
+
+                if len(entries) != 2:
+                    raise RuntimeError(
+                        "Calibration file should contain two space de-limited columns, "
+                        f"labelled as {self._DET_ID_LABEL} and {self._THETA_LABEL}"
+                    )
+
+                if not labels_checked:
+                    # The labels should be the first row in the file that doesn't begin with a #
+                    self._check_file_column_labels(entries)
+                    labels_checked = True
+                    continue
+
+                try:
+                    scanned_theta_offsets[int(entries[self._det_id_col_idx])] = float(entries[self._theta_offset_col_idx])
+                except ValueError:
+                    raise ValueError(
+                        "Invalid data in calibration file - detector ids should be integers and theta offsets should be floats"
```

Comment:
Yes, that's a good thought. I originally added this in because I had an issue when I put the labels in the wrong way round, but it could potentially just affect a single row.

Reply:
I'm going to print out the entry as part of the error message and see how the users get on with that for now, given that this error can come about a couple of ways. My understanding is that the users are planning to produce their calibration file via a script, so hopefully they should be unlikely to encounter this in practice.